megamek/src/megamek/client/ui/swing/AttackPhaseDisplay.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2023 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.swing;
+
+import megamek.common.actions.*;
+
+import java.util.*;
+
+import static megamek.client.ui.swing.util.UIUtil.guiScaledFontHTML;
+
+public abstract class AttackPhaseDisplay extends ActionPhaseDisplay {
+    // client list of attacks user has input
+    protected Vector<EntityAction> attacks;
+
+    protected AttackPhaseDisplay(ClientGUI cg) {
+        super(cg);
+    }
+
+    abstract protected void setDoneButtonValid();
```

Comment:
It would be good form to add a comment what an abstract method is supposed to be doing when overridden and/or when it is called.

Reply:
Nothing like writing docs to see how it could be simplified. I moved most of the repeated button managing into a single method on the ActionPhaseDisplay class. Now the child of classes handle the labeling and enabling of the Done and Skip buttons while the super class does the bookkeeping. This also makes it more reliable to reset the button labels and states after a change in client settings.